shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -810,13 +857,233 @@ bool OverrideGlobalPropertyFromIsolatedWorld(
+    args_array = v8::Array::New(isolate);
+  }
+
+  // Serialize the function
+  v8::Local<v8::String> serialized_func;
+  if (!func->FunctionProtoToString(isolate->GetCurrentContext())
+           .ToLocal(&serialized_func)) {
+    gin_helper::ErrorThrower(isolate).ThrowError(
+        "Failed to serialize function");
+    return v8::Local<v8::Value>();
+  }
+  v8::String::Utf8Value serialized_func_utf8(isolate, serialized_func);
+  std::string func_str(*serialized_func_utf8, serialized_func_utf8.length());
+
+  v8::Local<v8::Context> source_context = isolate->GetCurrentContext();
+  v8::Context::Scope source_scope(source_context);
+
+  // Get the target context
+  v8::MaybeLocal<v8::Context> maybe_target_context =
+      GetTargetContext(isolate, world_id);
+  v8::Local<v8::Context> target_context;
+  if (!maybe_target_context.ToLocal(&target_context)) {
+    isolate->ThrowException(v8::Exception::Error(gin::StringToV8(
+        isolate,
+        base::StringPrintf("Failed to get context for world %d", world_id))));
+    return v8::Local<v8::Value>();
+  }
+
+  // Compile the script
+  v8::MaybeLocal<v8::Script> maybe_compiled_script;
+  std::string error_message = "Unknown error during script compilation";
+  {
+    v8::Context::Scope target_scope(target_context);
+    v8::TryCatch try_catch(isolate);
+    std::string return_func_code = base::StringPrintf("(%s)", func_str.c_str());
+    maybe_compiled_script = v8::Script::Compile(
+        target_context, gin::StringToV8(isolate, return_func_code));
+    if (try_catch.HasCaught()) {
+      // Must throw outside of TryCatch scope
+      v8::String::Utf8Value error(isolate, try_catch.Exception());
+      error_message =
+          *error ? *error : "Unknown error during script compilation";
+    }
+  }
+  v8::Local<v8::Script> compiled_script;
+  if (!maybe_compiled_script.ToLocal(&compiled_script)) {
+    isolate->ThrowException(
+        v8::Exception::Error(gin::StringToV8(isolate, error_message)));
+    return v8::Local<v8::Value>();
+  }
+
+  // Run the script
+  v8::MaybeLocal<v8::Value> maybe_script_result;
+  {
+    v8::Context::Scope target_scope(target_context);
+    v8::TryCatch try_catch(isolate);
+    maybe_script_result = compiled_script->Run(target_context);
+    if (try_catch.HasCaught()) {
+      // Must throw outside of TryCatch scope
+      v8::String::Utf8Value error(isolate, try_catch.Exception());
+      error_message = *error ? *error : "Unknown error during script execution";
+    }
+  }
+  v8::Local<v8::Value> script_result;
+  if (!maybe_script_result.ToLocal(&script_result)) {
+    isolate->ThrowException(
+        v8::Exception::Error(gin::StringToV8(isolate, error_message)));
+    return v8::Local<v8::Value>();
+  }
+
+  // Get copied function from the script result
+  if (!script_result->IsFunction()) {
+    isolate->ThrowException(v8::Exception::Error(
+        gin::StringToV8(isolate, "Unknown error retrieving function")));
+    return v8::Local<v8::Value>();
```

Comment:
This is why MaybeLocal exists, we shouldn't be returning empty locals.

Reply:
Now returns `v8::Undefined(isolate)`